pytorch_lightning/trainer/train_loop_mixin.py

```diff
@@ -23,7 +23,15 @@ def train(self):
             # update training progress in trainer and model
             model.current_epoch = epoch_nb
             self.current_epoch = epoch_nb
-            self.total_batches = self.nb_training_batches + self.nb_val_batches
+
+            # val can be checked multiple times in epoch
+            is_val_epoch = (self.current_epoch + 1) % self.check_val_every_n_epoch == 0
+            val_checks_per_epoch = self.nb_training_batches // self.val_check_batch
```

Comment:
`self.val_check_batch` is a number of batches used for validation?

Reply:
Nope. It is the number of a training batch when validation loop should be run, i. e., we run validation loop in the end of the current training iteration if `(batch_nb + 1) % self.val_check_batch == 0`.

Reply:
I understand the `is_val_epoch` but still a bit hard to follow the remaining, why you divide  `nb_training_batches` (number of training batches in epoch) by `val_check_batch` (number of validation batches)...

Reply:
`val_check_batch` is not the number of validation batches. We run validation loop, when current training batch is equal to `val_check_batch`, `2 * val_check_batch`, `3 * val_check_batch` and so on. Thus during one training epoch we will run validation loop `nb_training_batches // val_check_batch` times. For example, if `val_check_batch = nb_training_batches` then we will run validation loop only once per epoch, at the very end of this epoch. And if `val_check_batch = 1` then we will run validation loop after every training batch, so there will be `nb_training_batches`

Reply:
so `val_check_batch` is the number of validation batches per epoch which are equally distributed

Reply:
I look at this from the other angle. By setting `nb_checks` you control how often you want to check your validation metrics. By setting `val_percent_check` you control how accurate your validation estimates. Ideally, you should set `val_percent_check=1` to obtain most accurate validation estimates. But in fact validation set may be quite big, so it would be very time consuming to use all validation data for one validation estimate. So you reduce `val_percent_check` to obtain less accurate but faster estimates.

So for me it is strange to distribute validation batches because total number of validation batches per epoch has no real meaning. Whereas `nb_checks` controls how often you make validation checks, and `val_percent_check` controls how accurate/fast your estimates.

Reply:
ok, correct me if I am wrong, just imagine that you have available 10 batches for validation but you use just 2 of them to be faster, and then you in some moment say I want to repeat it twice in each epoch... then you will repeat just the 2 (of 10) batches or will you take in fact another (4 of 10)?

Reply:
So answering your question:
> Just curious what is the reason running validation check of full size (eg 5 batches) multiple times in single epoch

Here is two things. You may want to run validation several times during one epoch if you have a lot of training data, so your epochs are very big but you want to check validation metrics more often. And you may want to compute these metrics using all validation data to obtain most accurate results.

Reply:
> ok, correct me if I am wrong, just imagine that you have available 10 batches for validation but you use just 2 of them to be faster, and then you in some moment say I want to repeat it twice in each epoch... then you will repeat just the 2 (of 10) batches or will you take in fact another (4 of 10)?

I guess it depends on how you have implemented `val_dataloader` :)
If you have set `shuffle=False`, then you will just repeat the same 2 batches. And if you have set  `shuffle=False` it depends on the implementation details which I don't know. If `val_dataloader` created before each validation loop, then first 2 batches and second 2 batches will be totally independent (it may be other 2 batches, or it may be the same two batches, or they can have only one common batch). And if `val_dataloader` is replaced only when exhausted, then you will take 4 of 10.